Source_Code/TM_UnitTests/TeamMentor.UnitTests/TeamMentor.UserData/Users/Test_Users_Creation.cs

```diff
@@ -408,6 +495,92 @@ public void CreateTmUserSigupResponse_Email_AlreadyExist()
             Assert.IsTrue(results.FirstOrDefault().Message == TMConsts.DEFAULT_SIGNUP_EMAIL_EXIST_MESSAGE);
 
         }
+
+        [Test]
+        public void CreateTmUserSigupResponse_Email_Address_Invalid_Bad_Format()
+        {
+            UserGroup.None.assert();                //  change current thread privildges to None
+            var newUser = new NewUser()
+            {
+                Username = "tA2!".add_RandomLetters(8),
+                Email = "notValidEmail"
+            };
+            var response = userData.createTmUserResponse(newUser);
+            Assert.IsTrue((response != null));
+            Assert.IsTrue(response != null && response.UserCreated ==0);
+            Assert.IsTrue(response != null && response.Signup_Status == Signup_Result.SignupStatus.Validation_Failed);
+            if (response != null)
+            {
+                var results = response.Validation_Results;
+                Assert.IsTrue(results.count() > 0);
+                Assert.IsTrue(results.FirstOrDefault().Field == "Email");
+                Assert.IsTrue(results.FirstOrDefault().Message == TMConsts.DEFAULT_EMAIL_ADDRESS_IS_INVALID);
+            }
+        }
+        [Test]
+        public void CreateTmUserSigupResponse_Email_Address_Large()
+        {
+            UserGroup.None.assert();                //  change current thread privildges to None
+            var newUser = new NewUser()
+            {
+                Username = "tA2!".add_RandomLetters(8),
+                Email = "".random_Email().add_RandomLetters(300)
+            };
+            var response = userData.createTmUserResponse(newUser);

```

Comment:
you can check the null here since ```assert_Null``` returns the object you tested.

so this can be

```csharp
var response = userData.createTmUserResponse(newUser).assert_Not_Null()
```